src/main/java/org/candlepin/config/ConfigProperties.java

```diff
@@ -285,6 +285,10 @@ private ConfigProperties() {
     // How long (in seconds) to wait for job threads to finish during a graceful Tomcat shutdown
     public static final String ASYNC_JOBS_THREAD_SHUTDOWN_TIMEOUT = "candlepin.async.thread.shutdown.timeout";
 
+    // How many days to allow a product to be orphaned before removing it on the next refresh or
+    // manifest import. Default: 30 days
+    public static final String ORPHANED_ENTITY_GRACE_PERIOD = "candlepin.refresh.orphan_entity_grace_period";
```

Comment:
I am unsure if we want this to be configurable. It would give customers a way to keep content that isn't covered by a subscription for too long or even indefinitely, bloating the database

Reply:
True enough, but the worst-case here is they turn it off and we're back to how 3.x and prior operate.